server/api/handlers/assignment_handler.go

```diff
@@ -36,9 +56,16 @@ func CreateAssignment(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusInternalServerError)
 	}
 
-	body.UserID = user.Email
+	body.UserID = user.ID
+
+	assignment, err := services.CreateAssignment(body)
+
+	if err != nil && services.IsDuplicatedKeyError(err) {
```

Comment:
If the error is not because of a duplicate key, but for example because something is missing, then according to this it will continue without returning an appropriate error message and the server will also say that the task was successfully saved (which did not happen...)